### PR TITLE
Encrypt relation's Internal Key in XLog

### PR DESCRIPTION
**Pull request by dAdAbird**
XLog record with its Internal Key for replication purposes. This commit makes this Key encrypted with the master key. Therefore we have to add also a master key name to XLog. That also means that standbys should have access to the very same master key as the primary.

Fix https://github.com/Percona-Lab/postgres-tde-ext/issues/56

### CODE REVIEW

File: src/access/pg_tde_tdemap.c [hunk 1]
```diff
@@ -68,45 +76,50 @@ pg_tde_create_key_fork(const RelFileLocator *newrlocator, Relation rel)
...
 	}
 
+	master_key_info = keyringGetLatestKey(MasterKeyName);
+	if(master_key_info == NULL)
+	{
+		master_key_info = keyringGenerateKey(MasterKeyName, INTERNAL_KEY_LEN);
+	}
+	if(master_key_info == NULL)
+	{
+        ereport(ERROR,
```
Comment (dutow) [inline src/access/pg_tde_tdemap.c:86]: Incorrect indentation - it's already there in the original, but now that we move it, it's a good time to fix it

File: src/access/pg_tde_tdemap.c [hunk 2]
```diff
@@ -394,6 +398,9 @@ pg_tde_xlog_create_fork(XLogReaderState *record)
 	char			*rec = XLogRecGetData(record);
 	RelFileLocator	rlocator;
 	InternalKey 	int_key = {0};
+	InternalKey 	int_key_enc = {0};
+	uint8 			mkey_len;
```
Comment (dutow) [inline src/access/pg_tde_tdemap.c:402]: seems like another indentation issue?

File: src/access/pg_tde_tdemap.c [hunk 3]
```diff
@@ -347,6 +330,27 @@ pg_tde_get_key_file_path(const RelFileLocator *newrlocator)
     return key_file_path;
 }
 
+void 
+pg_tde_decrypt_internal_key(const InternalKey *in, InternalKey *out, const char *MasterKeyName)
```
Comment (dutow) [inline src/access/pg_tde_tdemap.c:334]: `MasterKeyName` also seems inconsistent with the other variable/argument names

**Review by dutow — COMMENTED** [review]
Looks good, I only added a few formatting related comments

**Comment by dAdAbird** [comment]
I reckoned it make sense to run `pgindent` on the file

**Comment by dutow** [comment]
@dAdAbird pgIndent is a good idea, but could we do it in a separate PR instead? Maybe even on the entire code?

With this now, if you squash and merge, we'll end up with lots of formatting changes and some logic changes in a single commit.

**Review by dutow — APPROVED** [review]

**Review by EngineeredVirus — APPROVED** [review]
LGTM

**Pull request merged**